From 69e3a51b3dc50dd1697c64f7b2bcde54baa0a300 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Tue, 24 Jun 2025 11:27:51 +0200
Subject: [PATCH 221/293] winex11: Avoid requesting CWStackMode alone with
 managed windows.

Some window manager won't send a ConfigureNotify event if the stack
mode isn't actually changed, and we cannot predict whether it will.

We would be waiting forever for an event and delay other requests when
we can just request the above state the next time the window needs to
be reconfigured.
---
 dlls/winex11.drv/window.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index 726478ff040..f16a1c16096 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -1360,7 +1360,7 @@ static void window_set_config( struct x11drv_win_data *data, RECT rect, BOOL abo
     data->desired_state.rect = *new_rect;
     data->desired_state.above = above;
     if (!data->whole_window) return; /* no window, nothing to update */
-    if (EqualRect( old_rect, new_rect ) && (old_above || !above)) return; /* rects are the same, no need to be raised, nothing to update */
+    if (EqualRect( old_rect, new_rect ) && (old_above || !above || data->managed)) return; /* rects are the same, no need to be raised, nothing to update */
     if (window_needs_config_change_delay( data ))
     {
         TRACE( "window %p/%lx is updating _NET_WM_STATE/_MOTIF_WM_HINTS, delaying request\n", data->hwnd, data->whole_window );
-- 
2.43.0

